mesonbuild/mdevenv.py

```diff
@@ -169,7 +169,7 @@ def run(options: argparse.Namespace) -> int:
         if options.dump is True:
             dump(devenv, varnames, dump_fmt)
         else:
-            with open(options.dump, "w", encoding='utf-8') as output:
+            with open(os.path.join(workdir, options.dump), "w", encoding='utf-8') as output:
```

Comment:
What if `options.dump` is an absolute path?

Reply:
That works fine:
```
>>> os.path.join('/foo', '/bar')
'/bar'
```